pennylane_lightning/lightning_gpu/lightning_gpu.py

```diff
@@ -465,23 +475,23 @@ def _check_adjdiff_supported_measurements(measurements: List[MeasurementProcess]
                 )
 
             # The return_type of measurement processes must be expectation
-            if not all([m.return_type is Expectation for m in measurements]):
+            if any(m.return_type is not Expectation for m in measurements):
                 raise QuantumFunctionError(
                     "Adjoint differentiation method does not support expectation return type "
                     "mixed with other return types"
                 )
 
-            for m in measurements:
-                if not isinstance(m.obs, Tensor):
-                    if isinstance(m.obs, Projector):
+            for measurement in measurements:
+                if isinstance(measurement.obs, Tensor):
+                    if any(isinstance(o, Projector) for o in measurement.obs.non_identity_obs):
                         raise QuantumFunctionError(
-                            "Adjoint differentiation method does not support the Projector observable"
-                        )
-                else:
-                    if any([isinstance(o, Projector) for o in m.obs.non_identity_obs]):
-                        raise QuantumFunctionError(
-                            "Adjoint differentiation method does not support the Projector observable"
+                            "Adjoint differentiation method does not support the "
+                            "Projector observable"
                         )
+                elif isinstance(measurement.obs, Projector):
```

Comment:
Do we want this to be an elif, rather than an else?

Reply:
It should be elif for now, which is aligned with all other backends.